keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourcePersister.kt

```diff
@@ -58,30 +61,57 @@ class ResourcePersister(
         deliveryConfigRepository.store(it)
       }
 
-  fun upsert(resource: SubmittedResource<*>): Resource<out ResourceSpec> =
-    handlers.supporting(resource.apiVersion, resource.kind)
-      .normalize(resource)
+  fun <T : ResourceSpec> upsert(resource: SubmittedResource<*>): Resource<T> =
```

Comment:
This is the last and only place where a `SubmittedResource` can contain an un-parsed spec (i.e. just a `Map<String, Any?>`. I'm parsing it here and then methods in the repository, etc. use generic types based on `ResourceSpec`. This clarifies the responsibility of where this happens, stops using `normalize` unnecessarily (result was ignored except for name and it's called again later), and avoids some nasty glitches. Previously it was blowing up on updates because the parsed spec from `normalize` wasn't used.